packages/devtools_app/lib/src/shared/editor/api_classes.dart

```diff
@@ -501,6 +528,105 @@ class EditableArgumentsResult with Serializable {
   };
 }
 
+/// Constants for [CodeAction] prefixes used to filter the results returned by
+/// an [LspMethod.codeAction] request.
+abstract class CodeActionPrefixes {
+  static const flutterWrap = 'refactor.flutter.wrap';
+}
+
+/// The result of an [LspMethod.codeAction] request to the Analysis Server.
+///
+/// Contains a list of [CodeAction]s that can be performed.
+class CodeActionResult with Serializable {
+  CodeActionResult({required this.actions});
+
+  CodeActionResult.fromJson(List<Map<String, Object?>> list)
+    : this(actions: list.map(CodeAction.fromJson).toList());
+
+  final List<CodeAction> actions;
+
+  @override
+  Map<String, Object?> toJson() => {Field.actions: actions};
+}
+
+/// A code action (also known as a "Refactor" or "Quick Fix") that can be called
+/// via an [LspMethod.executeCommand] request.
+///
+/// For example, "Wrap with Center" or "Wrap with Container".
+class CodeAction with Serializable {
```

Comment:
nit: I would call this `CodeActionCommand` just to avoid some confusion (at least for me), because CodeActions according to LSP can actually be one of two things:

```
typedef CodeAction = CodeActionLiteral | Command
```

A `CodeActionLiteral` is a newer type that supports inline edits, but DTD clients do not support them, and therefore we should always be returning `Command`s to you here.

In analysis server code, I'm trying to consistently use `CodeAction` to mean the union of `CodeActionLiteral` and `Command`, and then those names explicitly where only one is allowed.